### PR TITLE
chore(RHTAPWATCH-1248): Provide admin user/pwd combo for ephemeral clusters

### PR DESCRIPTION
**Pull request by Kousalya1998**
- Add admin username and password in the credentials volume
- Add `APIserverURL` and `ConsoleURL` status field values from the CTI in the results

### CODE REVIEW

File: stepactions/eaas-get-ephemeral-cluster-credentials/0.1/eaas-get-ephemeral-cluster-credentials.yaml [hunk 1]
```diff
@@ -26,11 +26,23 @@ spec:
   results:
     - name: kubeconfig
       description: Relative path to the kubeconfig in the mounted volume
+    - name: username
+      description: Relative path to the username file in the mounted volume
+    - name: password
```
Comment (lcarva) [inline stepactions/eaas-get-ephemeral-cluster-credentials/0.1/eaas-get-ephemeral-cluster-credentials.yaml:31]: Just for the sake of avoiding heart attacks, can we rename this to something like "passwordPath"?

For consistency, `username` could also be renamed to `usernamePath` but it's probably ok to just record the username value directly in the result as that shouldn't be sensitive data.

Comment (Kousalya1998) [reply stepactions/eaas-get-ephemeral-cluster-credentials/0.1/eaas-get-ephemeral-cluster-credentials.yaml]: I updated it to `usernamePath` and `passwordPath`

File: stepactions/eaas-get-ephemeral-cluster-credentials/0.1/eaas-get-ephemeral-cluster-credentials.yaml [hunk 2]
```diff
@@ -56,3 +68,20 @@ spec:
...
+    echo "Wrote username to $CLUSTER_USERNAME"
+    echo "Wrote password to $CLUSTER_PASSWORD"
+    echo -n $(basename $CLUSTER_USERNAME) > $(step.results.usernamePath.path)
+    echo -n $(basename $CLUSTER_PASSWORD) > $(step.results.passwordPath.path)
+    
+    # Get `APIserverURL` and `ConsoleURL` status field values 
+    API_SERVER_URL=$("${OC[@]}" get cti $CLUSTER_NAME -o=jsonpath='{.status.apiServerURL}')
+    CONSOLE_URL=$("${OC[@]}" get cti $CLUSTER_NAME -o=jsonpath='{.status.consoleURL}')
+    echo -n "$API_SERVER_URL" > $(step.results.apiServerURL.path)
+    echo -n "$CONSOLE_URL" > $(step.results.consoleURL.path)
```
Comment (amisstea) [inline stepactions/eaas-get-ephemeral-cluster-credentials/0.1/eaas-get-ephemeral-cluster-credentials.yaml:86]: It would be nice to see these URLs output to the console as well.

**Review by amisstea — COMMENTED** [review]
Please don't merge this until @Kousalya1998 and I confirm it works properly.

File: stepactions/eaas-get-ephemeral-cluster-credentials/0.1/eaas-get-ephemeral-cluster-credentials.yaml (hunk 1 above)
Comment (amisstea) [reply stepactions/eaas-get-ephemeral-cluster-credentials/0.1/eaas-get-ephemeral-cluster-credentials.yaml]: I think we should keep `username` instead of `usernamePath` and emit it directly in the results.

Comment (Kousalya1998) [reply stepactions/eaas-get-ephemeral-cluster-credentials/0.1/eaas-get-ephemeral-cluster-credentials.yaml]: updated

File: stepactions/eaas-get-ephemeral-cluster-credentials/0.1/eaas-get-ephemeral-cluster-credentials.yaml (hunk 2 above)
Comment (Kousalya1998) [reply stepactions/eaas-get-ephemeral-cluster-credentials/0.1/eaas-get-ephemeral-cluster-credentials.yaml:88]: updated

File: stepactions/eaas-get-ephemeral-cluster-credentials/0.1/eaas-get-ephemeral-cluster-credentials.yaml (hunk 1 above)
Comment (amisstea) [reply stepactions/eaas-get-ephemeral-cluster-credentials/0.1/eaas-get-ephemeral-cluster-credentials.yaml]: @Kousalya1998 the only thing updated was the name of the param. The script is still writing this username to the credentials volume as the description states. The script should instead direct the value to the results.

Comment (Kousalya1998) [reply stepactions/eaas-get-ephemeral-cluster-credentials/0.1/eaas-get-ephemeral-cluster-credentials.yaml]: oh I didn't notice that it should emit it directly in the results. I think you want it to echo it in there

File: stepactions/eaas-get-ephemeral-cluster-credentials/0.1/eaas-get-ephemeral-cluster-credentials.yaml [hunk 3]
```diff
@@ -56,3 +64,18 @@ spec:
     "${OC[@]}" get secret $SECRET -o go-template --template="{{.data.kubeconfig|base64decode}}" > $CLUSTER_KUBECONFIG
     echo "Wrote kubeconfig to $CLUSTER_KUBECONFIG"
     echo -n $(basename $CLUSTER_KUBECONFIG) > $(step.results.kubeconfig.path)
+
+    # Get username and password
+    ADMIN_PASSWORD_SECRET=$("${OC[@]}" get cti $CLUSTER_NAME -o=jsonpath='{.status.adminPassword.name}')
+    echo "Found admin password secret: $ADMIN_PASSWORD_SECRET"
+    USERNAME=$("${OC[@]}" get secret $ADMIN_PASSWORD_SECRET -o jsonpath='{.data.username}' | base64 -d)
+    "${OC[@]}" get secret $ADMIN_PASSWORD_SECRET -o jsonpath='{.data.password}' | base64 -d > $CLUSTER_PASSWORD
```
Comment (amisstea) [inline stepactions/eaas-get-ephemeral-cluster-credentials/0.1/eaas-get-ephemeral-cluster-credentials.yaml:72]: Nit: Go templates make it possible to do the b64 decoding in one command (see `base64decode` above). It'd be nice to use one approach in the same script for consistency.

Comment (Kousalya1998) [reply stepactions/eaas-get-ephemeral-cluster-credentials/0.1/eaas-get-ephemeral-cluster-credentials.yaml]: @amisstea Makes sense. I updated it

**Review by amisstea — APPROVED** [review]
LGTM although there are some yaml linting errors to sort out still.

**Comment by Kousalya1998** [comment]
> LGTM although there are some yaml linting errors to sort out still.

checking that..

File: stepactions/eaas-get-ephemeral-cluster-credentials/0.1/eaas-get-ephemeral-cluster-credentials.yaml [hunk 4]
```diff
@@ -19,18 +19,26 @@ spec:
       description: A volume to which the remote cluster credentials will be written.
     - name: insecureSkipTLSVerify
       type: string
-      default: "false"
+      default: false
```
Comment (amisstea) [inline stepactions/eaas-get-ephemeral-cluster-credentials/0.1/eaas-get-ephemeral-cluster-credentials.yaml:22]: This line should not be changed. `false` is not a string unless it's quoted.

**Review by amisstea — APPROVED** [review]

**Comment by amisstea** [comment]
@Kousalya1998 your commits could use some cleanup. There should be just one.

File: task/build-vm-image/0.1/build-vm-image.yaml [hunk 5]
```diff
@@ -65,7 +65,7 @@ spec:
         name: varlibcontainers
   steps:
     - name: use-trusted-artifact
-      image: quay.io/redhat-appstudio/build-trusted-artifacts:latest@sha256:c3d440309b1a24cdfa8a2c89b668ca56952185f5c50e7abfa14b2aaafd5e2846
+      image: quay.io/redhat-appstudio/build-trusted-artifacts:latest@sha256:0379f5c2feec921a01300b386ce9414b10acfbc4499b187700d972b2360e1585
```
Comment (amisstea) [inline task/build-vm-image/0.1/build-vm-image.yaml:68]: @Kousalya1998 it looks like your rebase went awry. Please remove these unrelated changes.

**Review by amisstea — CHANGES_REQUESTED** [review]

File: task/build-vm-image/0.1/build-vm-image.yaml (hunk 5 above)
Comment (Kousalya1998) [reply task/build-vm-image/0.1/build-vm-image.yaml]: squashed it

Comment (Kousalya1998) [reply task/build-vm-image/0.1/build-vm-image.yaml]: when I rebased it again I dont see those unrelated changes now.

**Review by amisstea — APPROVED** [review]

**Comment by amisstea** [comment]
@chmeliik please review and merge if possible.
